src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonNuGetProject.cs

```diff
@@ -298,7 +298,7 @@ private JObject GetJson()
         {
             try
             {
-                using (var streamReader = new StreamReader(_jsonConfig.OpenRead()))
+                using (var streamReader = FileUtility.SafeRead(filePath: JsonConfigPath, read: (stream, filePath) => new StreamReader(stream)))
```

Comment:
This and the one above should create the JObject inside FileUtility.SafeRead.

 Just returning the unread stream could still hit issues and throw outside the retry loop.

The pattern should look the same as FileUtility.Replace where all the work is done inside